### PR TITLE
GroovyScript compat

### PR DESCRIPTION
**Pull request by brachy84**
Adds support fro GroovyScript. That includes:

- Bracket handlers
- Helper methods for recipe addition and removal
- Recipe reloading

Right now Crafttweaker support is untouched, but in the future we should yeet it, when proper tools for conversion exist.

### CODE REVIEW

File: src/main/java/gregtech/common/EventHandlers.java [hunk 1]
```diff
@@ -1,11 +1,14 @@
 package gregtech.common;
 
+import com.cleanroommc.groovyscript.sandbox.ScriptRunEvent;
```
Comment (ALongStringOfNumbers) [inline src/main/java/gregtech/common/EventHandlers.java:3]: Could you clean up these imports now that they are not used any more?

File: src/main/java/gregtech/api/recipes/map/Branch.java [hunk 2]
```diff
@@ -50,4 +50,13 @@ public Map<AbstractMapIngredient, Either<Recipe, Branch>> getSpecialNodes() {
...
     }
+
+    public void yeet() {
+        if (nodes != null) {
+            nodes.clear();
+        }
+        if (specialNodes != null) {
+            specialNodes.clear();
+        }
+    }
```
Comment (ALongStringOfNumbers) [inline src/main/java/gregtech/api/recipes/map/Branch.java:61]: This seems to be unused. Was there any plans for this at all?

File: src/main/java/gregtech/api/recipes/crafttweaker/CTRecipe.java [hunk 3]
```diff
@@ -22,6 +22,8 @@
 @SuppressWarnings("unused")
 public class CTRecipe {
 
+    // TODO YEET
```
Comment (ALongStringOfNumbers) [inline src/main/java/gregtech/api/recipes/crafttweaker/CTRecipe.java:25]: Were we going to remove CT compat? I don't remember that being the case.

Comment (brachy84) [reply src/main/java/gregtech/api/recipes/crafttweaker/CTRecipe.java:25]: CT is causing so many issues and shitty code. I would like its compate to be gone. Only when proper conversion tools are available of course.

**Review by ALongStringOfNumbers — APPROVED** [review]

**Pull request merged**